presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaPageSourceProvider.java

```diff
@@ -167,15 +169,20 @@ private InternalTable buildColumns(Session session, Set<QualifiedTablePrefix> pr
                             column.getExtraInfo(),
                             column.getComment());
                     ordinalPosition++;
+                    recordCnt++;
+                    if (limit.isPresent() && recordCnt == limit.getAsLong()) {
```

Comment:
I think we can simplify this by having `InternalTable.Builder` keep track of the size. Then we can add a method that takes `OptionalLong` to do the check:
```
if (table.atLimit(limit)) {
    return table.build();
}